File: TalkEasy-backend/routes/auth_routes.py
from fastapi import APIRouter
from controllers import auth_controller
from fastapi.responses import HTMLResponse

router = APIRouter(prefix="/auth", tags=["auth"])

router.post("/signup")(auth_controller.signup)
router.post("/login")(auth_controller.login)
router.post("/logout")(auth_controller.logout)
router.get("/me")(auth_controller.get_me)
router.get("/users")(auth_controller.get_all_users)
router.get("/user/{user_id}")(auth_controller.get_user_by_id)
router.get("/login/google")(auth_controller.auth_login_google)
router.get("/callback/google", response_class=HTMLResponse)(auth_controller.auth_callback_google)
router.post("/migrate-session")(auth_controller.migrate_session)

File: TalkEasy-backend/talkeasy-auth-service/src/routes/auth.routes.js
import express from 'express';
import passport from '../middleware/googleOAuth.middleware.js';
import { signup, login, logout, getMe, getAllUsers, getUserById, googleCallback } from '../controllers/auth.controller.js';
import { requireAuth } from '../middleware/auth.middleware.js';

const router = express.Router();

router.post('/signup', signup);
router.post('/login', login);
router.post('/logout', logout);
router.get('/me', requireAuth, getMe);
router.get('/users', requireAuth, getAllUsers);
router.get('/users/:user_id', requireAuth, getUserById);

// Google OAuth
router.get('/login/google', passport ? passport.authenticate('google', { scope: ['profile', 'email'] }) : (req, res) => res.status(501).json({ error: 'OAuth not configured' }));
router.get('/google/callback', passport ? passport.authenticate('google', { session: false }) : (req, res) => res.status(501).json({ error: 'OAuth not configured' }), googleCallback);

export default router;
